src/providers/api.service.js

```diff
@@ -103,6 +104,36 @@ function HereMapsAPIService($q, $http, HereMapsConfig, HereMapsUtilsService, Her
         return deferred.promise;
     }
 
+    function geocodeAddress(platform, params) {
+        if (!params)
+            return console.error('Missed required parameters');
+
+        var geocoder = platform.getGeocodingService(),
+            deferred = $q.defer(),
+            _params = {};
+
+        _params.gen = '8';
+
+        if (!!params.searchtext) {
+            // free text geocode
+            _params.searchtext = params.searchtext;
+        } else {
+            // partial address geocode
+            !!params.housenumber && (_params.housenumber = params.housenumber);
+            !!params.street && (_params.street = params.street);
+            !!params.city && (_params.city = params.city);
+            !!params.country && (_params.country = params.country);
```

Comment:
@aanev please do not check params existence step by step and just merge it instead of like:


```
var geocoder = platform.getGeocodingService(),
    deferred = $q.defer(),
    _params = {gen:8};

for (var key in params) { _params[key] = params[key]; }

geocoder.geocode(_params, function (response) {
    deferred.resolve(response)
}, function (error) {
    deferred.reject(error)
});


```